rest/class-boldgrid-backup-rest-archive.php

```diff
@@ -0,0 +1,277 @@
+		return $this->filter_schema_properties( $item );
+	}
+
+	/**
+	 * Create a new archive.
+	 *
+	 * For examples, please see rest/README.MD
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @see Boldgrid_Backup_Rest_Controller::permission_check for more information if you are getting
+	 *                                                        a 401 unauthorized error.
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return array                   Job Resource.
+	 */
+	public function create_item( $request ) {
+		// Initialize a new task.
+		$task = new Boldgrid_Backup_Admin_Task();
+		$task->init( [ 'type' => 'backup' ] );
+		$task->update();
+		$task->date_format = 'c';
+
+		// Trigger our backup.
+		$nopriv = new Boldgrid_Backup_Admin_Nopriv();
+		$nopriv->do_backup( [ 'task_id' => $task->get_id() ] );
+
+		return new WP_REST_Response( $task->get(), 200 );
+	}
+
+	/**
+	 * Get all archives for a WordPress.
+	 *
+	 * For examples, please see rest/README.MD
+	 *
+	 * @since X.X.X
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return array                   A collection of archive resources.
+	 */
+	public function get_items( $request ) {
+		// Init our list of raw backup data.
+		$core = apply_filters( 'boldgrid_backup_get_core', null );
+		$core->archives_all->init();
+
+		// Init our list of backups to return.
+		$backups = [];
+
+		foreach ( $core->archives_all->all as $backup ) {
+			$archive = Boldgrid\Backup\Archive\Factory::get_by_filename( $backup['filename'] );
+			$url     = $archive->generate_download_link( $backup['filename'] );
+
+			$backups[] = [
+				'id'            => $archive->get_id(),
+				'title'         => $archive->get_attribute( 'title' ),
+				'description'   => $archive->get_attribute( 'description' ),
+				'url'           => $url['download_url'],
+				'filename'      => $backup['filename'],
+				'creation_date' => date( 'c', $archive->timestamp ),
+			];
+		}
+
+		foreach ( $backups as &$backup ) {
+			$backup = $this->prepare_item_for_response( $backup, $request );
+		}
+
+		return $backups;
+	}
+
+	/**
+	 * Restore an archive.
+	 *
+	 * For examples, please see rest/README.MD
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return array                   Job Resource.
+	 */
+	public function restore( $request ) {
+		$id  = $request->get_param( 'id' );
+		$url = $request->get_param( 'url' );
+
+		// Initialize a new task.
+		$task = new Boldgrid_Backup_Admin_Task();
+		$task->init( [ 'type' => 'restore' ] );
+		$task->date_format = 'c';
+
+		if ( ! empty( $url ) || ! empty( $id ) ) {
+			// Update our task with either url or backup id.
+			if ( ! empty( $url ) ) {
+				$task->update_data( 'url', $url );
+			} else {
+				$task->update_data( 'backup_id', $id );
+			}
+
+			// Trigger our backup.
+			$nopriv = new Boldgrid_Backup_Admin_Nopriv();
+			$nopriv->do_restore( [ 'task_id' => $task->get_id() ] );
```

Comment:
Arrays should be declared with long array syntax per WordPress coding standards
https://developer.wordpress.org/coding-standards/wordpress-coding-standards/php/#declaring-arrays

Reply:
See similar comment above.